public/app/features/alerting/unified/components/settings/__mocks__/server.ts

```diff
@@ -32,31 +30,31 @@ export { history as alertmanagerConfigurationHistory };
 export const EXTERNAL_VANILLA_ALERTMANAGER_UID = 'vanilla-alertmanager';
 export const PROVISIONED_MIMIR_ALERTMANAGER_UID = 'provisioned-alertmanager';
 
-jest.spyOn(config, 'getAllDataSources');
-
-const mocks = {
-  getAllDataSources: jest.mocked(config.getAllDataSources),
-};
-
 export const mockDataSources = {
-  [EXTERNAL_VANILLA_ALERTMANAGER_UID]: mockDataSource<AlertManagerDataSourceJsonData>({
-    uid: EXTERNAL_VANILLA_ALERTMANAGER_UID,
-    name: EXTERNAL_VANILLA_ALERTMANAGER_UID,
-    type: DataSourceType.Alertmanager,
-    jsonData: {
-      implementation: AlertManagerImplementation.prometheus,
+  [EXTERNAL_VANILLA_ALERTMANAGER_UID]: mockDataSource<AlertManagerDataSourceJsonData>(
+    {
+      uid: EXTERNAL_VANILLA_ALERTMANAGER_UID,
+      name: EXTERNAL_VANILLA_ALERTMANAGER_UID,
+      type: DataSourceType.Alertmanager,
+      jsonData: {
+        implementation: AlertManagerImplementation.prometheus,
+      },
     },
-  }),
-  [PROVISIONED_MIMIR_ALERTMANAGER_UID]: mockDataSource<AlertManagerDataSourceJsonData>({
-    uid: PROVISIONED_MIMIR_ALERTMANAGER_UID,
-    name: PROVISIONED_MIMIR_ALERTMANAGER_UID,
-    type: DataSourceType.Alertmanager,
-    jsonData: {
-      // this is a mutable data source type but we're making it readOnly
-      implementation: AlertManagerImplementation.mimir,
+    { module: 'core:plugin/alertmanager' }
+  ),
+  [PROVISIONED_MIMIR_ALERTMANAGER_UID]: mockDataSource<AlertManagerDataSourceJsonData>(
+    {
+      uid: PROVISIONED_MIMIR_ALERTMANAGER_UID,
+      name: PROVISIONED_MIMIR_ALERTMANAGER_UID,
+      type: DataSourceType.Alertmanager,
+      jsonData: {
+        // this is a mutable data source type but we're making it readOnly
+        implementation: AlertManagerImplementation.mimir,
+      },
+      readOnly: true,
     },
-    readOnly: true,
-  }),
+    { module: 'core:plugin/alertmanager' }
```

Comment:
I'm wondering if we could add this by default when mocking alertmanager data source